### ISSUE TITLE
Ability to access the size/indices of individual records (for matrix-type records)

### ISSUE THREAD

**Issue by barrettk** [metrumresearchgroup/nmrec#32]
Using this method, we can access the `size` of each record:

```r
ctl <- nmrec::read_ctl(get_model_path(.mod))
pinfo <- nmrec:::create_param_index(ctl, "omega")
```
```r
> pinfo$details
[[1]]
[[1]]$type
[1] "diagonal"

[[1]]$subtype
[1] "implicit"

[[1]]$size
[1] 2

[[1]]$popts
...truncated...

[[2]]
[[2]]$type
[1] "block"

[[2]]$subtype
[1] "plain"

[[2]]$size
[1] 3

[[2]]$popts
...truncated...
```

In helping to address https://github.com/metrumresearchgroup/bbr/issues/640, we need a method for separating the full omega matrix into individual matrices (see this [comment](https://github.com/metrumresearchgroup/bbr/issues/640#issuecomment-1908817551)). The full procedure for addressing the issue is outlined in this [comment](https://github.com/metrumresearchgroup/bbr/issues/640#issuecomment-1910692616). As you can see in the setup section however, I hard coded the individual matrices; shown again below:
```r
omega_full <- get_initial_est(.mod)$omegas
omega_1 <- omega_full[1:2, 1:2]
omega_2 <- omega_full[3:5, 3:5]
```

There are a number of ways I could get those dimensions. 
 - The simplest method I can think of from an `nmrec` perspective, would be to have a helper to pull out the `size` attribute, and then I can determine the indices on the `bbr` front. I know we dont want to export `create_param_index` for good reason, so a helper that extracts some of these attributes could be more ideal.
    - A more ideal helper function could be one that either A) returns the matrix indices of the record, or B) completely handles the separation of the full matrix by record. I assume the latter doesnt really make sense to have in `nmrec`, but just putting it out there.
- Make the individual matrices (matrix per OMEGA/matrix-type record) an attribute in `extract_omega`/`extract_sigma`.
- Some other solution

In terms of how and what information we would return from the `nmrec` side, I would 100% leave that up to you, based on what you think `nmrec` should be responsible for. In the end, I just need a method for separating the matrices using exported `nmrec` functions, and the minimum information I need to do that, is how many rows/columns each record spans, along with the order they show up in in the control stream file (i.e. to determine the indices the record spans within the full compiled matrix).

**Comment by kyleam** [comment]
@barrettk Thanks for posting this issue.

> The simplest method I can think of from an nmrec perspective, would be to have a helper to pull out the size attribute, and then I can determine the indices on the bbr front [...]

I agree that that's the simplest given the current implementation.  In hopes that this doesn't block you, I've pushed a scratch implementation to [scratch/extra-param-sizes](https://github.com/metrumresearchgroup/nmrec/compare/main...scratch/extra-param-sizes).

I'm undecided on where we should end up eventually, but it probably makes sense to wait until we have a better idea of how bbr#640 settles.

**Comment by barrettk** [comment]
Awesome thanks! Will give it a shot and let you know how it goes

**Comment by barrettk** [comment]
Hey @kyleam While the original intention for this feature was for matrix handling (see linked issue above), I also made use of it for [tabulating the record number](https://github.com/metrumresearchgroup/bbr/pull/635#issuecomment-1917561152) in the new `initial_estimates()` function.  

The matrix handling, along with tweaking initial omegas and sigmas altogether, will be handled in a separate PR.

However, after discussing with @seth127, we agreed it would be nice to include this metadata (`record_number` column) in this [PR](https://github.com/metrumresearchgroup/bbr/pull/635) (Note that it wont be that exact PR, as Im going to start with a clean branch). This new column is potentially distinguishing non-informative priors, which is in line with some of the [other adjustments](https://github.com/metrumresearchgroup/bbr/blob/6c1aaadfbebe8d8a7927572d6cdc00b91c1bd8d8/R/initial-estimates.R#L218-L255) we've made.
